src/Kafka/ManageKafkaPermissions/components/SelectAccount.tsx

```diff
@@ -0,0 +1,194 @@
+import React, { useState } from "react";
+import { useTranslation } from "react-i18next";
+import { Account, PrincipalType } from "../types";
+import {
+  Divider,
+  Select,
+  SelectGroup,
+  SelectOption,
+  SelectVariant,
+  ValidatedOptions,
+  SelectProps,
+  SelectOptionObject,
+} from "@patternfly/react-core";
+
+export type Validated<T> = {
+  value: T;
+  validated?: ValidatedOptions;
+  errorMessage?: string;
+};
+
+export type SelectAccountProps = {
+  id: string | undefined | SelectOptionObject;
```

Comment:
```suggestion
  value: string | undefined
```

Naming is hard, I know. But `id` in HTML has its own meaning. A common way to provide the value of a field is via a prop called `value`. I'd use that. 

Also, why should the parent deal with the complexities of the PF's Select component? The idea of writing specialized components is to make it easier to work with the problem at hand. If an account is a string, we should talk about strings. Not the weird `string | undefined | SelectOptionObject` that comes from the PF's Select

Reply:
I'll fix the naming :) . I had a question about the type, I've posted it [here](https://github.com/redhat-developer/app-services-ui-components/pull/175#discussion_r820959213)